src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -644,6 +644,15 @@ def create_vault(cmd, client,  # pylint: disable=too-many-locals
                  no_self_perms=None,
                  tags=None,
                  no_wait=False):
+    from azure.core.exceptions import HttpResponseError
+    try:
+        vault = client.get(resource_group_name=resource_group_name, vault_name=vault_name)
+        if vault:
+            raise InvalidArgumentValueError('The specified vault: {} already exists'.format(vault_name))
```

Comment:
In my perspective, forbidding `create` from being called multiple times is against the Azure API design:

https://docs.microsoft.com/en-us/azure/architecture/microservices/design/api-design#idempotent-operations

> The HTTP specification states that GET, PUT, and DELETE methods must be idempotent.

**Vaults - Create Or Update** is a PUT operation:

https://docs.microsoft.com/en-us/rest/api/keyvault/vaults/create-or-update

```
PUT https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}?api-version=2019-09-01
```

This PR introduces BREAKING CHANGE and we'd better call that out. This script works before but now it breaks:

```
az keyvault create -n myvault -g testrg
az keyvault create -n myvault -g testrg
``` 

Only my 2 cents.

Reply:
Sure, I'll add the breaking change notes in history